src/colorizer/Collection.ts

```diff
@@ -307,25 +342,89 @@ export default class Collection {
    * Collection representing its contents (either the loaded collection or a dummy collection
    * containing just the dataset).
    * @param url the URL resource to attempt to load.
-   * @param fetchMethod optional override for the fetch method.
+   * @param options optional configuration object containing any of the following properties:
+   *  - `fetchMethod` optional override for the fetch method.
+   *  - `reportWarning` optional callback for reporting warning messages during loading.
    * @throws an error if `url` is not a URL.
    * @returns a Promise of a new Collection object, either loaded from a collection JSON file or
    * generated as a wrapper around a single dataset.
    */
-  public static async loadFromAmbiguousUrl(url: string, fetchMethod = fetchWithTimeout): Promise<Collection> {
+  public static async loadFromAmbiguousUrl(
+    url: string,
+    options: Partial<{ fetchMethod: typeof fetchWithTimeout; reportWarning: ReportWarningCallback }> = {}
+  ): Promise<Collection> {
     // TODO: Also handle Nucmorph URLs that are pasted in? If website base URL matches, redirect?
 
     if (!isUrl(url)) {
       throw new Error(`Provided resource '${url}' is not a URL and cannot be loaded.`);
     }
 
+    let collectionLoadError: Error | null = null;
+    let datasetLoadError: Error | null = null;
     try {
-      return await Collection.loadCollection(url, fetchMethod);
+      return await Collection.loadCollection(url, options);
     } catch (e) {
+      collectionLoadError = e as Error;
+      console.warn(e);
       console.log("URL resource could not be parsed as a collection; attempting to make a single-database collection.");
     }
 
     // Could not load as a collection, attempt to load as a dataset.
-    return await Collection.makeCollectionFromSingleDataset(url);
+    try {
+      const collection = Collection.makeCollectionFromSingleDataset(url);
+      // Attempt to load the default dataset immediately to surface any loading errors.
+      const loadResult = await collection.tryLoadDataset(collection.getDefaultDatasetKey());
+      if (!loadResult.loaded) {
+        throw new Error(loadResult.errorMessage);
+      }
+      return collection;
+    } catch (e) {
+      datasetLoadError = e as Error;
+    }
+
+    // Assumption: if both loads failed and the file matches one of the default filenames,
+    // show only the relevant load error.
+    if (url.endsWith(DEFAULT_COLLECTION_FILENAME)) {
+      throw collectionLoadError;
+    } else if (url.endsWith(DEFAULT_DATASET_FILENAME)) {
+      throw datasetLoadError;
+    }
+
```

Comment:
here I'm a little confused, is there a case where neither of the two if statements on 387 and 389 are true? I'm just wondering why you're throwing errors and then also checking what's in the error message.

Reply:
Yeah, anytime we're given a directory instead of a specific filename.

For example, 
`https://allencell.s3.amazonaws.com/aics/nuc-morph-dataset/timelapse_feature_explorer_datasets/full-interphase_dataset` is ambiguous because it could either be pointing to the directory of a collection (in which case we check for `collection.json`) or a dataset's directory (in which case we check for `dataset.json`). In this case, if both error out, we don't know what the user wanted us to load.

Somthing like `https://allencell.s3.amazonaws.com/aics/nuc-morph-dataset/timelapse_feature_explorer_datasets/full-interphase_dataset/my-custom-file.json` would also be ambiguous, and we would have to try loading it as both a collection and a dataset.